implementation of r-squared

shark8me opened the pull request:
Implementation of coefficient of determination or R-squared. 

- filters rows in which both pairs are valid numbers
- return nil for NaNs, and invalid inputs (e.g. no valid pairs of inputs).

mikera commented:
Thoughts:

Should the argument order be (actual, predicted?) I think this is more logical

Filtering is expensive! Some implementation don't have nils so it would be good to have a fast path. I can see a few options:

1. Check the element-type of the array and run faster code if nils are not possible (i.e. for primitive backed arrays)
2. Have an optional argument `filter-nils?` if people want this functionality
3. Have a separate function that detects nils (but is slower)
4. Create a protocol that implementations can optionally implement if they have a fast way of doing this

I think option 2 is probably best, but happy to discuss

shark8me commented:
Updated the branch with

- argument order changed to actual, predicted (similar to [sklearn](http://scikit-learn.org/stable/modules/generated/sklearn.metrics.r2_score.html))
- added an optional argument to filter nils if required.
